### PR TITLE
[msal-angular] replace minimatch

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR:
- Removes `minimatch` and `path` from msal-angular dependencies
- Replaces minimatch usage with `StringUtils.matchPattern` from msal-browser
- Updates documentation on wildcard matching
- Improves interceptor tests related to wildcard matching
- IE11 compatibility added to angular11-sample-app

Changes made to msal-browser are mirrored on #2693.

### CODE REVIEW

File: lib/msal-angular/docs/v2-docs/initialization.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ export class AppModule {}
 
 Using MsalInterceptor is optional and you can write your own interceptor if you choose to. Alternatively, you can also explicitly acquire tokens using the acquireToken APIs.
 
-As of `@azure/msal-angular@2`, `protectedResourceMap` supports wildcard patterns that are supported by [minimatch](https://github.com/isaacs/minimatch). `unprotectedResources` is deprecated and no longer an option for configuration. Instead, setting a scope value of `null` on a resource will prevent it from getting tokens.
+As of `@azure/msal-angular@2`, `protectedResourceMap` supports using `*` for wildcards. `unprotectedResources` is deprecated and no longer an option for configuration. Instead, setting a scope value of `null` on a resource will prevent it from getting tokens.
```
Comment (jasonnutter) [inline lib/msal-angular/docs/v2-docs/initialization.md:92]: It would be good to call this out in the upgrade guide also (in addition to the fact we no longer support just putting your root domain in the protected resource map, must use wildcard matching)

Comment (jo-arroyo) [reply lib/msal-angular/docs/v2-docs/initialization.md:92]: Done.

File: lib/msal-angular/package.json [hunk 2]
```diff
@@ -19,10 +19,7 @@
   },
   "main": "./dist/bundles/azure-msal-angular.umd.js",
   "typings": "./dist/azure-msal-angular.d.ts",
-  "dependencies": {
-    "minimatch": "^3.0.4",
-    "path": "^0.12.7"
-  },
+  "dependencies": {},
```
Comment (hectormmg) [inline lib/msal-angular/package.json:22]: Nit: Maybe remove the dependencies key?

**Review by hectormmg — APPROVED** [review]
LGTM!

File: lib/msal-angular/package.json (hunk 2 above)
Comment (jo-arroyo) [reply lib/msal-angular/package.json:22]: Good point! Thank you!

File: lib/msal-angular/src/msal.guard.ts [hunk 3]
```diff
@@ -79,9 +79,8 @@ export class MsalGuard implements CanActivate {
         /*
          * If a page with MSAL Guard is set as the redirect for acquireTokenSilent,
          * short-circuit to prevent redirecting or popups.
-         * TODO: Update to allow running in iframe once allowRedirectInIframe is implemented
          */
-        if (UrlString.hashContainsKnownProperties(window.location.hash) && BrowserUtils.isInIframe()) {
+        if (UrlString.hashContainsKnownProperties(window.location.hash)) {
```
Comment (jasonnutter) [inline lib/msal-angular/src/msal.guard.ts:83]: Sorry, I wasn't clear about the changes here. We want to keep the `isInIframe` check, but only if `allowRedirectInIframe` is `false`.
```suggestion
        if (UrlString.hashContainsKnownProperties(window.location.hash) && BrowserUtils.isInIframe() && !allowRedirectInIframe) {
```

I think for now I would revert this change, and we can talk about how to handle this incrementally, since may require some other work to expose the `allowRedirectInIframe` value.

Comment (jo-arroyo) [reply lib/msal-angular/src/msal.guard.ts]: Okay makes sense. Reverting

**Review by tnorling — APPROVED** [review]
Looks good, just a friendly reminder to also make relevant changes to the dev branch :)

**Comment by jasonnutter** [comment]
> Looks good, just a friendly reminder to also make relevant changes to the dev branch :)

Since this is just for Angular v2, I don't think a separate PR is needed for dev.

**Comment by tnorling** [comment]
> > Looks good, just a friendly reminder to also make relevant changes to the dev branch :)
> 
> Since this is just for Angular v2, I don't think a separate PR is needed for dev.

This contains a new export from msal-browser, need to make sure that is available in the next release

**Comment by jasonnutter** [comment]
> > > Looks good, just a friendly reminder to also make relevant changes to the dev branch :)
> > 
> > 
> > Since this is just for Angular v2, I don't think a separate PR is needed for dev.
> 
> This contains a new export from msal-browser, need to make sure that is available in the next release

Ah right, missed that.

**Comment by jo-arroyo** [comment]
> Looks good, just a friendly reminder to also make relevant changes to the dev branch :)

Thanks for the reminder, @tnorling !
